networking_nsxv3/db/db.py

```diff
@@ -167,11 +158,14 @@ def get_port(context, port_id):
         PortBinding.host,
         PortBinding.vif_details,
         trunk_model.Trunk.port_id
+    ).select_from(
+        Port
     ).filter(
-        Port.id == port_id
+        Port.id == port_id,
+        PortBindingLevel.host == host
```

Comment:
You probably meanth PortBinding.host, because joining with PortBindingLevel joins with all levels of this host. (Causing to return an excessive amount of ports)

Reply:
Fixed